src/basic/env-file.c

```diff
@@ -225,10 +211,13 @@ static int parse_env_file_internal(
                 case DOUBLE_QUOTE_VALUE_ESCAPE:
                         state = DOUBLE_QUOTE_VALUE;
 
-                        if (!strchr(NEWLINE, c)) {
+                        if (c == '\"')
+                                value[n_value++] = c;
+                        else if (!strchr(NEWLINE, c)) {
                                 if (!GREEDY_REALLOC(value, value_alloc, n_value+2))
```

Comment:
So I think you need `n_value+3` here, since `n_value+2` was right for a single character and now you have two characters here...